src/main/java/org/jabref/logic/importer/fileformat/EndnoteXmlImporter.java

```diff
@@ -28,42 +26,75 @@
+            Map.entry(StandardEntryType.TechReport, "Report"),
+            Map.entry(StandardEntryType.Unpublished, "Manuscript"),
+            Map.entry(StandardEntryType.InProceedings, "Conference Paper"),
+            Map.entry(StandardEntryType.Conference, "Conference"),
+            Map.entry(IEEETranEntryType.Patent, "Patent"),
+            Map.entry(StandardEntryType.Online, "Web Page"),
+            Map.entry(IEEETranEntryType.Electronic, "Electronic Article"),
+            Map.entry(StandardEntryType.Misc, "Generic")
+    );
+
+    private static final Map<Field, String> FIELD_MAPPING = Map.ofEntries(
+            Map.entry(StandardField.TITLE, "title"),
+            Map.entry(StandardField.AUTHOR, "authors"),
+            Map.entry(StandardField.EDITOR, "secondary-authors"),
+            Map.entry(StandardField.JOURNAL, "full-title"),
+            Map.entry(StandardField.BOOKTITLE, "secondary-title"),
+            Map.entry(StandardField.EDITION, "edition"),
+            Map.entry(StandardField.SERIES, "tertiary-title"),
+            Map.entry(StandardField.VOLUME, "volume"),
+            Map.entry(StandardField.NUMBER, "number"),
+            Map.entry(StandardField.ISSUE, "issue"),
+            Map.entry(StandardField.PAGES, "pages"),
+            Map.entry(StandardField.LOCATION, "pub-location"),
+            Map.entry(StandardField.CHAPTER, "section"),
+            Map.entry(StandardField.HOWPUBLISHED, "work-type"),
+            Map.entry(StandardField.PUBLISHER, "publisher"),
+            Map.entry(StandardField.ISBN, "isbn"),
+            Map.entry(StandardField.ISSN, "issn"),
+            Map.entry(StandardField.DOI, "electronic-resource-num"),
+            Map.entry(StandardField.URL, "web-urls"),
+            Map.entry(StandardField.FILE, "pdf-urls"),
+            Map.entry(StandardField.ABSTRACT, "abstract"),
+            Map.entry(StandardField.KEYWORDS, "keywords"),
+            Map.entry(StandardField.PAGETOTAL, "page-total"),
+            Map.entry(StandardField.NOTE, "notes"),
+            //  Map.entry(StandardField.LABEL, "label"), // We omit this field
+            Map.entry(StandardField.LANGUAGE, "language"),
+            // Map.entry(StandardField.KEY, "foreign-keys"),  // We omit this field
+            Map.entry(StandardField.ADDRESS, "auth-address")
+    );
+
     private final ImportFormatPreferences preferences;
+
     private final XMLInputFactory xmlInputFactory;
 
     public EndnoteXmlImporter(ImportFormatPreferences preferences) {
         this.preferences = preferences;
         xmlInputFactory = XMLInputFactory.newInstance();
-        // prevent xxe (https://rules.sonarsource.com/java/RSPEC-2755)
-        // not suported by aalto-xml
-        // xmlInputFactory.setProperty(XMLConstants.ACCESS_EXTERNAL_SCHEMA, "");
-        // required for reading Unicode characters such as &#xf6;
-        xmlInputFactory.setProperty(XMLInputFactory.IS_COALESCING, true);
```

Comment:
Did you remove that statement on purpose?

Reply:
Oh, I missed this. Basically, I was having a hard time extending the old importer so I started writing from scratch. In that, I did not think about needing this.

Reply:
Then, please re-add this. Also the lines above including "not supported by aalto-xml". Someone in future should have a look at that one.

Reply:
Okay, done.